### PR TITLE
DPC-1387 Feature: Delete IP via `DELETE /v2/system/{system_id}/ip/{id}`

### PR DESCRIPTION
**Pull request by charmcitygavin**
### Fixes [DPC-1387](https://jira.cms.gov/browse/DPC-1387)

Adds SSAS v2 endpoint: DELETE /v2/system/[id]/ip/[id]

### Proposed Changes

* Adds SSAS v2 endpoint: DELETE /v2/system/[id]/ip/[id]
* Adds unit test for endpoint

### Change Details

An authenticated admin user may now use this endpoint to soft-delete an IP address associated with a system.

### Security Implications

- [ ] new software dependencies

- [ ] security controls or supporting software altered

- [ ] new data stored or transmitted

- [ ] security checklist is completed for this change

- [ ] requires more information or team discussion to evaluate security implications

- [x] no PHI/PII is affected by this change

### Acceptance Validation

The endpoint works as expected in Postman when testing manually. There is also a unit test.

Two IP addresses available on a system:

Screen Shot 2021-06-09 at 10 40 29 AM

One IP targeted for deletion:

Screen Shot 2021-06-09 at 10 42 28 AM

On the next GET request, only the other IP is returned:

Screen Shot 2021-06-09 at 10 43 06 AM

### Feedback Requested

Please hit me with any and all feedback, especially regarding the unit tests. If you see anything I could simplify or expand, please call it out! 😄

### CODE REVIEW

**Comment by MrMorie** [comment]
Looks like the DELETE method is returning the deleted resource, and "deleted_at" is still null.

Typically for a delete a 204 status and an empty body is appropriate,
But if we do return the entity I think it should have the "deleted_at" status

**Comment by charmcitygavin** [comment]
Great point @MrMorie. Thanks! I've updated it to return a 204 with empty body on successful deletion. 👍🏻

**Review by MrMorie — APPROVED** [review]
LGTM!

Before merging you want to run it by:
https://bcda-ci.adhocteam.us/job/BCDA%20-%20Build%20and%20Package/build

Just to make sure CI works and all quality stages pass.

File: ssas/service/admin/api.go [hunk 1]
```diff
@@ -499,6 +499,48 @@ func getSystemIPs(w http.ResponseWriter, r *http.Request) {
...
+	}
+
+	ipID := chi.URLParam(r, "id")
+
+	trackingID := uuid.NewRandom().String()
+	ssas.OperationCalled(ssas.Event{Op: "DeleteSystemIP", TrackingID: trackingID, Help: "calling from admin.deleteSystemIP()"})
+
+	err = system.DeleteIP(ipID, trackingID)
+	if err != nil {
+		//ssas.OperationFailed
```
Comment (dhgreene) [inline ssas/service/admin/api.go:536]: On line 532, if you create a reference to the `ssas.Event` you can reuse it here. Then we can uncomment. 😄

**Review by MrMorie — APPROVED** [review]
LGTM

**Comment by MrMorie** [comment]
@charmcitygavin I merged all 3 of my PRs and that introduced some merge conflicts with yours. I went ahead and resolved them.

File: ssas/service/admin/api.go (hunk 1 above)
Comment (charmcitygavin) [reply ssas/service/admin/api.go]: Thanks David! I updated with a reference to the event and also made tracking ID generation more consistent across the file. 👍🏻

**Review by dhgreene — APPROVED** [review]
💯

**Pull request merged**